sdk/src/main/scala/com/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -77,6 +78,41 @@ class AccountForgeMessageBuilder(mainchainSynchronizer: MainchainSynchronizer,
     }
   }
 
+  def calculateBaseFee(nodeView: View, parentId: BlockId): BigInteger = {
+    // If the current block is the first block, return the InitialBaseFee.
+    if (nodeView.history.getCurrentHeight < 2) {
+      return Account.INITIAL_BASE_FEE
+    }
+
+    nodeView.history.getBlockById(parentId).asScala match {
+    case None => Account.INITIAL_BASE_FEE
+    case Some(block) =>
+      val blockHeader = block.header
+      val gasTarget = blockHeader.gasLimit / 2
+
+        // If the parent gasUsed is the same as the target, the baseFee remains unchanged
+      if (blockHeader.gasUsed == gasTarget) {
+        return blockHeader.baseFee
+      }
+
+      if (blockHeader.gasUsed > gasTarget) {
+        // If the parent block used more gas than its target, the baseFee should increase by 12.5%
```

Comment:
The comment is a bit misleading, the 12.5% is the maximum change.

Reply:
Yes, I already saw that and changed